lib/search/cubit/search_cubit.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:bloc/bloc.dart';
+import 'package:fluttercon/search/cubit/search_state.dart';
+import 'package:injectable/injectable.dart';
+import '../../../common/data/models/failure.dart';
+import '../../../common/repository/api_repository.dart';
+import '../../../common/repository/db_repository.dart';
+import '../models/search_result.dart';
+
+@injectable
+class SearchCubit extends Cubit<SearchState> {
+  final ApiRepository _apiRepository;
+  final DBRepository _dbRepository;
+
+  SearchCubit(
+      this._apiRepository,
+      this._dbRepository,
+      ) : super(const SearchInitial());
+
+  Future<void> search(String query) async {
+    emit(const SearchLoading());
+    try {
+      final sessions = await _dbRepository.fetchSessions();
+      final speakers = await _dbRepository.fetchSpeakers();
+      final sponsors = await _dbRepository.fetchSponsors();
+      final organizers = await _dbRepository.fetchOrganisers();
```

Comment:
As a performance showcase, instead of loading all the database records into memory to perform a search, what you can do instead is delegate the search to the local database engine Isar. Ref: https://isar.dev/recipes/full_text_search.html

So what you can do is just load all the search results here instead then format them into search results for the UI. 

For example:
Approach 1: (New db repo methods)
```dart
final sessions = await _dbRepository.searchSessions(query:query);
final speakers = await _dbRepository.searchSpeakers(query:query);
....
```
then return results

Approach 2: (Add nullable params)
```dart
final sessions = await _dbRepository.fetchSessions(query: query);

// Signature change to
Future<List<LocalSession>> fetchSessions({
    BookmarkStatus? bookmarkStatus,
    String? sessionLevel,
    String? sessionType,
    String? query,
  })

// Then tweak the algo with
.optional(
          query != null,
          (q) => q.titleContains(query!),
        )
// to return the search results to the cubit
```

then once they are back in the cubit, you can format them as you have here and return to the widget.

Reply:
Fixed